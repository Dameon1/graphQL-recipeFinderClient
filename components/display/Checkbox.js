
import React from 'react';
import '../styles/checkbox.css';

export default class Checkbox extends React.Component {
  state = {
    isChecked: false,
  }

  toggleCheckboxChange = () => {
    const { handleCheckboxChange, label } = this.props;
    this.setState(({ isChecked }) => (
      { isChecked: !isChecked, }
    ));
    handleCheckboxChange(label);
  }

  render() {
    const { label } = this.props;
    const { isChecked } = this.state;

    return (
      <div className="checkbox" onClick={this.toggleCheckboxChange}>
        <label>
          <input
            type="checkbox"
            value={label}
            checked={isChecked}           
          />
          {label}
        </label>
      </div>
    );
  };
};
