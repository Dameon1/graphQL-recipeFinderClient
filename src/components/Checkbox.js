import React from "react";
import { IngredientBox } from "./styles";

export class Checkbox extends React.Component {
  state = {
    isChecked: false
  };

  toggleCheckboxChange = () => {
    const { handleCheckboxChange, label } = this.props;
    this.setState(({ isChecked }) => ({ isChecked: !isChecked }));
    handleCheckboxChange(label);
  };

  render() {
    const { label } = this.props;
    const { isChecked } = this.state;

    return (
      <IngredientBox onClick={this.toggleCheckboxChange}>
        <label>
          <input
            type="checkbox"
            value={label}
            checked={isChecked}
            onChange={this.toggleCheckboxChange}
          />
          {label}
        </label>
      </IngredientBox>
    );
  }
}

export default Checkbox;