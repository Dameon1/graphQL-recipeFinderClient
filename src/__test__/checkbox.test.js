//Checkbox is used iteratively in the recipe search for to display ingredient choices

import React from "react";
import { shallow, mount } from "enzyme";
import Checkbox from "../components/display/Checkbox";

describe("<Checkbox />", () => {
  it("renders without crashing", () => {
    shallow(<Checkbox />);
  });

  it("displays the correct ingredient with the same value", () => {
    const handleCheckboxChange = jest.fn();
    const label = "Milk";
    let wrapper = mount(
      <Checkbox label={label} handleCheckboxChange={handleCheckboxChange} />
    );
    const input = wrapper.find("input");
    expect(wrapper.text()).toEqual(label);
    expect(wrapper.length).toEqual(1);
    expect(input.props().value).toEqual(label);
  });

  it("fires a function on click and changes state", () => {
    const handleCheckboxChange = jest.fn();
    const label = "Milk";
    let wrapper = mount(
      <Checkbox label={label} handleCheckboxChange={handleCheckboxChange} />
    );
    expect(wrapper.state("isChecked")).toEqual(false);
    //simulate click and change state
    wrapper.find("input").simulate("click");
    expect(handleCheckboxChange).toHaveBeenCalledWith(label);
    expect(wrapper.state("isChecked")).toEqual(true);
    //simulate click and change state
    wrapper.find("input").simulate("click");
    expect(handleCheckboxChange).toHaveBeenCalledWith(label);
    expect(wrapper.state("isChecked")).toEqual(false);
  });
});
