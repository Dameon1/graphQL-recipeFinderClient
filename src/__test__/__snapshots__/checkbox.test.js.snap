// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> matches the snapshot 1`] = `
<Checkbox
  handleCheckboxChange={[Function]}
  label="Milk"
>
  <styled.label
    onClick={[Function]}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-kgoBCf",
            "isStatic": true,
            "lastClassName": "fDUXOo",
            "rules": Array [
              "
          display: inline-block;
          width: 50%;
          padding: 2px;
          font-size: 22px;
        
          :hover {
            border: dotted #454545 1px;
          }
        ",
            ],
          },
          "displayName": "styled.label",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-kgoBCf",
          "target": "label",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
    >
      <label
        className="sc-kgoBCf fDUXOo"
        onClick={[Function]}
      >
        <label>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
            value="Milk"
          />
          Milk
        </label>
      </label>
    </StyledComponent>
  </styled.label>
</Checkbox>
`;
